test/evergreen/find_cmake.sh

```diff
@@ -17,7 +17,14 @@ find_cmake ()
     elif [ -f "/opt/mongodbtoolchain/v4/bin/cmake" ]; then
         CMAKE="/opt/mongodbtoolchain/v4/bin/cmake"
         CTEST="/opt/mongodbtoolchain/v4/bin/ctest"
+    elif [ -f "/opt/homebrew/bin/cmake" ]; then
+        # Mac: If a homebrew version of CMake and CTest is installed then it is likely newer than the default one.
+        # This is important on M1 Macs as M1 silicon is only supported in CMake 3.19.2 and later
+        # (see https://cmake.org/cmake/help/latest/release/3.19.html), and the default version may be too old.
+        CMAKE="/opt/homebrew/bin/cmake"
+        CTEST="/opt/homebrew/bin/ctest"
```

Comment:
I'm curious whether toolchain cmake is available for use on the Evergreen macOS distros. 
This change will definitely be helpful for local macOS development.

Reply:
The find_cmake.sh script looks first for cmake in the toolchain location (/opt/mongodbtoolchain/v4/bin/cmake) but that isn't found, so the script then looks in /opt/homebrew/bin/cmake, and then /Applications/CMake.app/Contents/bin/cmake (which is where in the Intel Mac install finds version 3.11).
So, I'm not sure we can improve this at the moment.